packages/edge-bundler/node/bundler.ts

```diff
@@ -106,6 +106,10 @@ export const bundle = async (
   const userFunctions = userSourceDirectories.length === 0 ? [] : await findFunctions(userSourceDirectories)
   const internalFunctions = internalSrcFolder ? await findFunctions(internalSrcFolders) : []
   const functions = [...internalFunctions, ...userFunctions]
+
+  // Early exit when there are no functions to bundle
+  if (functions.length === 0) return {}
```

Comment:
this results in a breaking change to the signature of the `bundle()` function exported from this package. if that's intentional, we should mark this PR as breaking (though note that since it changes two packages it'll cut a major for both).

(btw personal opinion but this is why I like https://typescript-eslint.io/rules/explicit-function-return-type/. it forces changes to return types to be explicit and visible.)

Reply:
Yeah, I think it make sense try not to touch edge-bundler and in particular change signature if possible here. I did take a look at how we handle this for serverless and we have similar early bail there, but it is in step itself instead of bundler/zisi and applied similar change in https://github.com/netlify/build/pull/6659 which is alternative to this PR - no breaking changes there and no additional file lookups